core/types/transaction_signing_rip7560.go

```diff
@@ -29,17 +29,24 @@ func (s rip7560Signer) Hash(tx *Transaction) common.Hash {
 		tx.Type(),
 		[]interface{}{
 			s.chainId,
+			aatx.Nonce,
 			tx.GasTipCap(),
 			tx.GasFeeCap(),
 			tx.Gas(),
-			tx.Data(),
 			tx.AccessList(),
 
 			aatx.Sender,
+			// no AuthorizationData here - this is hashing "for signing"
```

Comment:
its a random order. please use same ordering as TX hash (ignore authorizationData which should be last in txHash)